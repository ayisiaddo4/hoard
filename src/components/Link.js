import React from 'react';
import PropTypes from 'prop-types';
import {
  View,
  Linking,
  Image,
  StyleSheet,
  TouchableOpacity,
} from 'react-native';
import Icon from 'components/Icon';
import NavigatorService from 'lib/navigator';
import T from 'components/Typography';

import memoize from 'lodash/fp/memoize';

const navigateTo = memoize((to) => () => NavigatorService.navigate(to));
const externalLinkTo = memoize((to) => () => {
  Linking.openURL(to).catch(err => {
    if (__DEV__) {
      // eslint-disable-next-line no-console
      console.error('An error occurred', err);
    }
  });
});

export default function Link({title, to, external, icon}) {
  const onPress = external ? externalLinkTo(to) : navigateTo(to);
  return (
    <TouchableOpacity style={styles.container} onPress={onPress}>
      <View style={styles.leftSide}>
        {icon && <Image source={icon} style={styles.icon}/>}
        <T.Light style={styles.title}>{title}</T.Light>
      </View>
      <Icon icon="ios-arrow-forward" style={{size: 10, marginLeft: 'auto'}} />
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    paddingVertical: 20,
    borderBottomWidth: 1,
    borderBottomColor: 'rgba(151, 151, 151, 0.21)'
  },
  leftSide: {
    flex: 1,
    flexDirection: 'row',
    alignItems:'center',
  },
  icon: {
    width: 20,
    height: 20,
    resizeMode: 'contain',
  },
  title: {
    marginLeft: 20,
    color: 'white'
  },
});

Link.propTypes = {
  external: PropTypes.bool,
  icon: Image.propTypes.source,
  title: PropTypes.string,
  to: PropTypes.string,
};
