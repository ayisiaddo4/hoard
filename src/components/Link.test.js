/* eslint-env jest */

import React from 'react';
import Link from './Link';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(<Link />).toJSON();
  expect(tree).toMatchSnapshot();
});
