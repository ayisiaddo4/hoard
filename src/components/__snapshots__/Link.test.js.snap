// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderBottomColor": "rgba(151, 151, 151, 0.21)",
      "borderBottomWidth": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "opacity": 1,
      "paddingVertical": 20,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "marginRight": 20,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#17181e",
            "fontFamily": "HelveticaNeue",
            "fontSize": 14,
            "letterSpacing": 0,
          },
          Object {
            "color": "white",
          },
        ]
      }
    />
  </View>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontSize": 10,
        },
        undefined,
        Object {
          "fontFamily": "Ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    
  </Text>
</View>
`;
